tests/test_mutations.py

```diff
@@ -936,6 +937,91 @@ def verify_model_ts(self, ts, model, discrete, verify_roots):
             p /= sum(p)
             self.chisquare(row, sum(row) * p)
 
+    def verify_model_ts_general(self, ts, model, discrete, verify_roots, mutation_rate):
+        alleles = [a.decode() for a in model.alleles]
+        num_alleles = len(alleles)
+        roots = np.zeros((num_alleles,))
+
+        if discrete:
+            Q = model.transition_matrix.copy()
+            np.fill_diagonal(Q, 0.0)
+            row_sums = Q.sum(axis=1)
+            np.fill_diagonal(Q, -1 * row_sums)
+            observed_transitions = np.zeros((num_alleles, num_alleles))
+            expected = np.zeros((num_alleles, num_alleles))
+
+            for v in ts.variants(samples=range(ts.num_nodes), impute_missing_data=True):
+                t = ts.at(v.position)
+                for n in t.nodes():
+                    pn = t.parent(n)
+                    if pn != tskit.NULL:
+                        pa = v.alleles[v.genotypes[pn]]
+                        da = v.alleles[v.genotypes[n]]
+                        observed_transitions[alleles.index(pa), alleles.index(da)] += 1
+                        expected += _matrix_exponential(
+                            t.branch_length(n) * mutation_rate * Q
+                        )
+
+            np.fill_diagonal(observed_transitions, 0.0)
+            np.fill_diagonal(expected, 0.0)
+
+            for j, (row, p) in enumerate(zip(observed_transitions, expected)):
+                p[j] = 0
+                p /= sum(p)
+                self.chisquare(row, sum(row) * p)
+
+        else:
+            observed_transitions = np.zeros((num_alleles, num_alleles))
+            for mut in ts.mutations():
+                if mut.parent == tskit.NULL:
```

Comment:
if we're not doing discrete mutations, the mutation parents should always be NULL?